src/runtime/TemporalObject.cpp

```diff
@@ -327,6 +328,135 @@ TemporalObject::DateTime TemporalObject::parseTemporalDateTimeString(ExecutionSt
     return TemporalObject::parseValidIso8601String(state, isoString);
 }
 
+std::map<std::string, int> TemporalObject::parseTemporalDurationString(ExecutionState& state, const std::string& isoString)
+{
+    int sign = 1;
+    unsigned int index = 0;
+
+    int years = -1;
+    int months = -1;
+    int weeks = -1;
+    int days = -1;
+    int hours = -1;
+    int minutes = -1;
+    int seconds = -1;
+    int fHours = -1;
+    int fMinutes = -1;
+    int fSeconds = -1;
+
+    if (isoString[index + 1] != 'P') {
```

Comment:
What happens if `isoString` is an empty string? This might be a buffer overread.